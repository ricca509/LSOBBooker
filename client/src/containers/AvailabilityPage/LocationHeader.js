import React, { Component } from 'react';
import { observer, inject } from 'mobx-react';
import moment from 'moment';

const getStartDate = (dates) => {
  return dates ?
    (<div>
      <strong>From </strong>{ moment(dates.start).format('dddd, MMMM Do YYYY') }
    </div>) :
    (<div><strong>From </strong>-</div>);
}

const getEndDate = (dates) => {
  return dates ?
    (<div>
      <strong>to </strong>{ moment(dates.end).format('dddd, MMMM Do YYYY') }
    </div>) :
    (<div><strong>to </strong>-</div>);
}

export const LocationHeader = ({
  locationId,
  serviceId,
  dates,
  onSelectedLocationChange,
  onSelectedServiceChange,
  onResetClick,
  store
}) => {
  if (!locationId) {
    const locations = Object.keys(store.locationList)
      .map(loc => (
        <option value={loc} key={loc}>{store.locationList[loc]}</option>
      ));

    return (
      <select name="select" onChange={onSelectedLocationChange}>
        <option value="">Select location</option>
        { locations }
      </select>
    );
  }

  if (locationId && !serviceId) {
    const services = Object.keys(store.serviceList[locationId])
      .map(svc => (
        <option value={svc} key={svc}>{store.serviceList[locationId][svc]}</option>
      ));

    return (
      <div>
        <div><strong>Location </strong>{ store.locationList[locationId] }</div>
        <select name="select" onChange={onSelectedServiceChange}>
          <option value="">Select service</option>
          { services }
        </select>
      </div>
    )
  }

  if (locationId && serviceId) {
    return (
      <main>
        <div>
          <div>
            <strong>Location </strong>{ store.locationList[locationId] }
          </div>
          <div>
            <strong>Service </strong>{ store.serviceList[locationId][serviceId] }
          </div>
          <div>
            <a
              className="button button-small button-clear"
              onClick={onResetClick}>
              Change location/service
            </a>
          </div>
        </div>
        <div>
          { getStartDate(dates) }
          { getEndDate(dates) }
        </div>
      </main>
    );
  }
}

LocationHeader.displayName = 'LocationHeader';

export default inject('store')(observer(LocationHeader));
