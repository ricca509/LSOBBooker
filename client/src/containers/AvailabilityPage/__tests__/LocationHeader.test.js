import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import moment from 'moment';

import LocationHeader from '../LocationHeader';

describe('LocationHeader', () => {
  it('renders the location intended for the search', () => {
    const wrapper = shallow(<LocationHeader
      location='far far away'
      dates={{
        start: moment('12-25-2016', 'MM-DD-YYYY'),
        end: moment('12-29-2016', 'MM-DD-YYYY')
      }}
      service='haircut'
    />);

    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it('renders without dates if no date is passed', () => {
    const wrapper = shallow(<LocationHeader
      location='far far away'
      service='haircut'
      dates={null}
    />);

    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
