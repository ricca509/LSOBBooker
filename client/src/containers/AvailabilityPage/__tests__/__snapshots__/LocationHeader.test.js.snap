exports[`LocationHeader when location and service are selected renders the selected location and service 1`] = `
<div>
  <div>
    <strong>
      Location 
    </strong>
    location1
  </div>
  <div>
    <strong>
      Service 
    </strong>
    service1
  </div>
  <div>
    <strong>
      From 
    </strong>
    Sunday, December 25th 2016
  </div>
  <div>
    <strong>
      to 
    </strong>
    Thursday, December 29th 2016
  </div>
</div>
`;

exports[`LocationHeader when location and service are selected renders without dates if no date is passed 1`] = `
<div>
  <div>
    <strong>
      Location 
    </strong>
    location1
  </div>
  <div>
    <strong>
      Service 
    </strong>
    service1
  </div>
  <div>
    <strong>
      From 
    </strong>
    -
  </div>
  <div>
    <strong>
      to 
    </strong>
    -
  </div>
</div>
`;

exports[`LocationHeader when location is selected and service is not renders selected location and service selection 1`] = `
<div>
  <div>
    <strong>
      Location 
    </strong>
    location1
  </div>
  <select
    name="select">
    <option
      value="">
      Select service
    </option>
    <option
      value="service1">
      service1
    </option>
    <option
      value="service2">
      service2
    </option>
  </select>
</div>
`;

exports[`LocationHeader when no location or service is selected renders location selection 1`] = `
<select
  name="select">
  <option
    value="">
    Select location
  </option>
  <option
    value="location1">
    location1
  </option>
  <option
    value="location2">
    location2
  </option>
</select>
`;
